sdk/servicebus/service-bus/src/serializers/ruleResourceSerializer.ts

```diff
@@ -0,0 +1,431 @@
+        "xmlns:p4": "http://www.w3.org/2001/XMLSchema-instance"
+      };
+    } else {
+      let isSqlFilter: boolean = false;
+
+      const givenFilter: any = rule.filter as any;
+
+      if (givenFilter.sqlExpression != undefined) {
+        isSqlFilter = true;
+      }
+
+      if (isSqlFilter) {
+        const sqlFilter: SqlFilter = rule.filter as SqlFilter;
+        resource.Filter = {
+          SqlExpression: sqlFilter.sqlExpression,
+          Parameters: getRawSqlParameters(sqlFilter.sqlParameters),
+          CompatibilityLevel: 20,
+          RequiresPreprocessing: getStringOrUndefined(sqlFilter.requiresPreprocessing)
+        };
+        resource.Filter[Constants.XML_METADATA_MARKER] = {
+          "p4:type": "SqlFilter",
+          "xmlns:p4": "http://www.w3.org/2001/XMLSchema-instance"
+        };
+      } else {
+        const correlationFilter: CorrelationFilter = rule.filter as CorrelationFilter;
+
+        resource.Filter = {
+          CorrelationId: correlationFilter.correlationId,
+          Label: correlationFilter.label,
+          To: correlationFilter.to,
+          ReplyTo: correlationFilter.replyTo,
+          ReplyToSessionId: correlationFilter.replyToSessionId,
+          ContentType: correlationFilter.contentType,
+          SessionId: correlationFilter.sessionId,
+          MessageId: correlationFilter.messageId,
+          Properties: correlationFilter.userProperties
+        };
+        resource.Filter[Constants.XML_METADATA_MARKER] = {
+          "p4:type": "CorrelationFilter",
+          "xmlns:p4": "http://www.w3.org/2001/XMLSchema-instance"
+        };
+      }
+    }
+
+    if (rule.action == undefined || rule.action.sqlExpression == undefined) {
+      // Defaults to creating an empty rule action instance if none specified
+      resource.Action = {};
+      resource.Action[Constants.XML_METADATA_MARKER] = {
+        "p4:type": "EmptyRuleAction",
+        "xmlns:p4": "http://www.w3.org/2001/XMLSchema-instance"
+      };
+    } else {
+      resource.Action = {
+        SqlExpression: rule.action.sqlExpression,
+        Parameters: getRawSqlParameters(rule.action.sqlParameters),
+        CompatibilityLevel: 20,
+        RequiresPreprocessing: getStringOrUndefined(rule.action.requiresPreprocessing)
+      };
+      resource.Action[Constants.XML_METADATA_MARKER] = {
+        "p4:type": "SqlRuleAction",
+        "xmlns:p4": "http://www.w3.org/2001/XMLSchema-instance"
+      };
+    }
+
+    return serializeToAtomXmlRequest("RuleDescription", resource);
+  }
+
+  async deserialize(response: HttpOperationResponse): Promise<HttpOperationResponse> {
+    return deserializeAtomXmlResponse(["TopicName", "SubscriptionName", "RuleName"], response);
+  }
+}
+
+enum SqlParameterType {
+  Integer = "l28:int",
+  String = "l28:string",
+  Long = "l28:long",
+  Date = "l28:date"
+}
+/**
+ * Represents type of SQL `Parameter` in ATOM based management operations
+ */
+export type SqlParameter = {
```

Comment:
`SqlParameter` is not used in any other file. So, it need not be exported

Reply:
This is meant for users to use for constructing and passing in inputs